sched/scheduler/jobRunner.go

```diff
@@ -0,0 +1,267 @@
+package scheduler
+
+import (
+	"fmt"
+	"github.com/scootdev/scoot/saga"
+	"github.com/scootdev/scoot/sched"
+	cm "github.com/scootdev/scoot/sched/clustermembership"
+	dist "github.com/scootdev/scoot/sched/distributor"
+	"sync"
+)
+
+const MAX_RETRY = 3
+
+type jobRunner struct {
+	job       sched.Job
+	saga      saga.Saga
+	sagaState *saga.SagaState
+	dist      *dist.PoolDistributor
+
+	updateChan chan stateUpdate
+	wg         sync.WaitGroup
+}
+
+func NewJobRunner(job sched.Job, saga saga.Saga, sagaState *saga.SagaState, nodes []cm.Node) jobRunner {
+	jr := jobRunner{
+		job:        job,
+		saga:       saga,
+		sagaState:  sagaState,
+		dist:       dist.NewPoolDistributor(cm.StaticClusterFactory(nodes)),
+		updateChan: make(chan stateUpdate),
```

Comment:
I want this to be done in order to ensure saga log messages are logged before I start a task.

i.e. StartTask must be logged before task is executed to ensure correctness of the protocol in recovery.  otherwise you could pass the update (it doesn't get logged) then StartTask and then a failure occurs so the message never gets logged.  I made the buffer declaration explicit